### ISSUE TITLE
train val test split for single illuminant and mixed illuminant training schedules

### ISSUE THREAD

**Issue by CemEntok** [DY112/LSMI-dataset#15]
Hello

is there any JSON file including train/val/test splits you can share publicly so that your single illuminant and mixed illuminant results can be reproduced?
I noticed that the files you shared in separate_json_files are only consisting of multi-illuminant images.

**Comment by DY112** [comment]
Hello. There is a slight discrepancy between the number of instances we used to generate the numbers reported in the paper at the time of publication and the results our current code produces.

The single illuminant results reported in the paper used sub-image (_1) of the 99 two-illuminant test scene, and the multi-illuminant used the two/three-illuminant test scene (_12,_123).
The code has now been updated to handle all the sub-images (_1,_12,_13) of the three-illuminant scene as well.

If you want to generate slightly reduced test instances to reproduce our result, please refer to following code:
https://github.com/DY112/LSMI-dataset/blob/4e12d52ad6a17a0b7350113fb1f287263ac096ec/2_preprocess_data.py#L47C1-L48C25

This is how we initially processed the test scene, and it generates a test set that matches what is reported in the paper.

**Comment by CemEntok** [comment]
Thank you for your reply!

1. So your results are obtained from the test set consisting of:

- (_1) of 2-illuminant scenes/places: a total of 99 images (A)
- (_12) of 2-illuminant scenes/places: a total of 99 images (B)
- (_123) of 3-illuminant scenes/places: a total of 13 images (C)

So by looking at these bullet-ins, you test the model with data A for obtaining single illuminant results, data B+C for obtaining multi-illuminant results, and data A+B+C for obtaining mixed-illuminant results so that the numbers shared in the paper are matched:

![image](https://github.com/DY112/LSMI-dataset/assets/97525722/46f97d4b-8508-409d-a11f-fb2cec9e95e3)

Am I approaching the testing of the model in your paper in the right manner? Correct me please if I am wrong.

----

2. Did you train the model with two_iilum_train and three_illum_train places and then test the model with the test mentioned above or did you change the training set for every test case such as separating the training set for each case of single illuminant, multi-illuminant and mixed illuminant cases?

---

3. The last point is, that you did not include _1, _12, _13 of the 3-illuminant scenes in the results of the paper. Therefore, if I had included them in the training, I could not reproduce the results I suppose.

**Comment by DY112** [comment]
Sorry for late reply.
1. You are right.
2. Every evaluation results are from single model which is trained with full train set.
3. You are right. We provide option flag whether to include sub-images of 3-illuminant scenes or not.
